### DISCUSSION TITLE
How to list a projects robot accounts using the API?

### DISCUSSION THREAD

**Discussion by dpavlotzky** [goharbor/harbor#18743] [Q&A] [answered]
Hi,

I am trying to list the project level robot accounts using the api, but I keep getting an empty array:

`curl -u ‘admin:*********’ -X GET ‘https://<harbor_url>/api/v2.0/robots?level=project&project_id=1’`
`curl -u ‘admin:*********’ -X GET ‘https://<harbor_url>/api/v2.0/robots?q=level=project,project_id=1’`

Response:

`[]`

Listing the system level robot accounts work fine:

`curl -u ‘admin:********’ -X GET ‘https://<harbor_url>/api/v2.0/robots?level=system’`
`curl -u ‘admin:********’ -X GET ‘https://<harbor_url>/api/v2.0/robots?q=level=system’`

Logs show nothing weird, even with log_level debug.
Any ideas?

**Comment by AllForNothing** [comment]
You need to specify the level and project id like this:
`curl -u ‘admin:*********’ -X GET ‘https://<harbor_url>/api/v2.0/robots?q=Level=project&ProjectID=1&page_size=15&page=1`

**Reply by dpavlotzky** [reply]
That doesn’t work, but gives a `400 Bad Request`.

**Reply by dpavlotzky** [reply, marked as answer]
Ah, I think I found it: project level robot account has been deprecated (https://github.com/goharbor/harbor/pull/15296). Only system level robot accounts remain and project access can be assigned to these robot accounts. I was thrown off by the fact that the UI allows me to create project level robot accounts, but these account don’t show up with the API.
